### PR TITLE
Updated serializer configuration and made them lazy

### PR DESCRIPTION
**Pull request by lfgcampos**
The goal for this PR is to do the same we did on Framework, making Serializers lazy and only initializing them if the user doesn't provide one.

It has an added benefit of not always initializing XStream.
For making it work, the initialization of the `messageConverter` was also pushed to the end of the builder call.

I am not sure we have meaningful tests to be added in this case.

### CODE REVIEW

**Comment by gklijs** [comment]
This is expected to be a whole or part of the fix for https://github.com/AxonFramework/AxonFramework/issues/2184

File: kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/streamable/StreamableKafkaMessageSource.java [hunk 1]
```diff
@@ -153,12 +155,21 @@ private String buildConsumerGroupId() {
...
         private ConsumerFactory<K, V> consumerFactory;
         private Fetcher<K, V, KafkaEventMessage> fetcher;
-        @SuppressWarnings({"unchecked", "squid:S1874"})
-        private KafkaMessageConverter<K, V> messageConverter =
-                (KafkaMessageConverter<K, V>) DefaultKafkaMessageConverter.builder()
-                                                                          .serializer(XStreamSerializer.defaultSerializer())
-                                                                          .build();
+        private KafkaMessageConverter<K, V> messageConverter;
         private Supplier<Buffer<KafkaEventMessage>> bufferFactory = SortedKafkaMessageBuffer::new;
+        private Supplier<Serializer> serializer;
```
Comment (gklijs) [inline kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/streamable/StreamableKafkaMessageSource.java:160]: I think it would be better to just set the default here already. Then we also don't need to check for null, the XStream serializer will already log if the code is called.

File: kafka-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfigurationIntegrationTest.java [hunk 2]
```diff
@@ -555,7 +564,9 @@ protected static class TestConfiguration {
 
         @Bean
         public Serializer eventSerializer() {
-            return XStreamSerializer.defaultSerializer();
+            return XStreamSerializer.builder()
```
Comment (gklijs) [inline kafka-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfigurationIntegrationTest.java:567]: Just wondering why this change? Just so it won't log for using the default and/or because it's deprecated?

**Review by gklijs — CHANGES_REQUESTED** [review]

File: kafka-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfigurationIntegrationTest.java (hunk 2 above)
Comment (lfgcampos) [reply kafka-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/kafka/autoconfig/KafkaAutoConfigurationIntegrationTest.java:567]: Since I was changing serializers, I get rid of the deprecated method and migrated it to the 'new way'. Tests should still be the same as before, just a code smell less =)

**Review by gklijs — APPROVED** [review]
Looks good now, thanks 👍

File: kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/streamable/StreamableKafkaMessageSource.java [hunk 3]
```diff
@@ -153,12 +155,23 @@ private String buildConsumerGroupId() {
...
-                                                                          .serializer(XStreamSerializer.defaultSerializer())
-                                                                          .build();
+        private KafkaMessageConverter<K, V> messageConverter;
         private Supplier<Buffer<KafkaEventMessage>> bufferFactory = SortedKafkaMessageBuffer::new;
+        private Supplier<Serializer> serializer = () -> XStreamSerializer.builder()
+                                                                         .xStream(new XStream(new CompactDriver()))
+                                                                         .build();
+
+        /**
+         * Sets the {@link Serializer} used to serialize and deserialize messages.
```
Comment (smcvb) [inline kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/streamable/StreamableKafkaMessageSource.java:165]: It would be friendly if this little bit of Javadoc also shared what the default serializer is.

File: kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/subscribable/SubscribableKafkaMessageSource.java [hunk 4]
```diff
@@ -210,13 +213,24 @@ public void close() {
...
-                                                                          .build();
+        private KafkaMessageConverter<K, V> messageConverter;
         private boolean autoStart = false;
         private int consumerCount = 1;
+        private Supplier<Serializer> serializer = () -> XStreamSerializer.builder()
+                                                                         .xStream(new XStream(new CompactDriver()))
+                                                                         .build();
+
+        /**
+         * Sets the {@link Serializer} used to serialize and deserialize messages.
```
Comment (smcvb) [inline kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/consumer/subscribable/SubscribableKafkaMessageSource.java:224]: It would be friendly if this little bit of Javadoc also shared what the default serializer is.

File: kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisher.java [hunk 5]
```diff
@@ -248,14 +251,25 @@ public void shutDown() {
...
+        private KafkaMessageConverter<K, V> messageConverter;
         private MessageMonitor<? super EventMessage<?>> messageMonitor = NoOpMessageMonitor.instance();
         private TopicResolver topicResolver = m -> Optional.of(DEFAULT_TOPIC);
         private long publisherAckTimeout = 1_000;
+        private Supplier<Serializer> serializer = () -> XStreamSerializer.builder()
+                                                                         .xStream(new XStream(new CompactDriver()))
+                                                                         .build();
+
+        /**
+         * Sets the {@link Serializer} used to serialize and deserialize messages.
```
Comment (smcvb) [inline kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisher.java:263]: It would be friendly if this little bit of Javadoc also shared what the default serializer is.

**Review by smcvb — APPROVED** [review]
Missing a little bit of JavaDoc. Otherwise, this is good to go if you ask me.

**Review by sandjelkovic — APPROVED** [review]

**Review by smcvb — APPROVED** [review]
Still think this looks fine.
